nlp/query.js

```diff
@@ -145,13 +147,27 @@ async function query(req, res) {
     if (exact !== null)
         result = exact.map((code) => ({ code, score: 'Infinity' })).concat(result);
 
+
     applyCompatibility(result, thingtalk_version);
     res.set("Cache-Control", "no-store,must-revalidate");
-    res.json({
-        candidates: result,
-        tokens: tokens,
-        entities: tokenized.entities
+    classifier.classify(query).then((value) => {
```

Comment:
`query` is an asynchronous function, but you're using `Promise`-style code (explicit `.then`) here.
Either `return` the promise or use `await`, otherwise error handling will be incorrect.